### PR TITLE
Feature/EX-5384: Add setSnippetConfig function in the API

### PR DESCRIPTION
**Pull request by guillei10**
EX-5384

Adding a `setSnippetConfig` function in the API in order to modify the `snippetConfig`. The implementation is found in the `XInstaller` but it gets exposed in the API throughout a facade. Also:

- Had to change the payload in the `ExtraParamsInitialized` to be a clone of the object instead. This avoids having this reference in the store, and we can change the `extraParams` properties without `Vuex` complaining.
- Changed the priority of how the `SnippetConfigExtraParams` adds properties in the `extraParams` object.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [x] `Main`
- [ ] Other. Specify:

## How has this been tested?

Tests performed according to [testing guidelines](./contributing/tests.md): 

## Checklist:

- [x] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [x] I have performed a **self-review** on my own code.
- [x] I have **commented** my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [x] My changes generate **no new warnings**.
- [x] I have added **tests** that prove my fix is effective or that my feature works.
- [x] New and existing **unit tests pass locally** with my changes.

### CODE REVIEW

File: packages/x-components/src/x-installer/api/__tests__/default-api.spec.ts [hunk 1]
```diff
@@ -28,4 +31,32 @@ describe('testing default X API', () => {
     defaultXAPI.addProductToCart(productId);
     expect(listener).toHaveBeenCalledWith(productId);
   });
+
+  it('changes the `SnippetConfig` when calling the `setSnippetConfig` function', async () => {
```
Comment (javieri-empathy) [inline packages/x-components/src/x-installer/api/__tests__/default-api.spec.ts:35]: I'm missing another test to ensure that non initial values get added reactively. For example, in this current test add the `store` attribute in another step.

**Review by javieri-empathy — CHANGES_REQUESTED** [review]

File: packages/x-components/src/x-installer/x-installer/x-installer.ts [hunk 2]
```diff
@@ -296,13 +306,14 @@ export class XInstaller {
   ): Vue | undefined {
     if (this.options.app !== undefined) {
       const vue = this.getVue();
-      snippetConfig = vue.observable(snippetConfig);
+      this.snippetConfig = vue.observable(snippetConfig);
+      const snippetConfigObservable = this.snippetConfig;
```
[the remark this answers is not in the document]
Comment (guillei10) [reply packages/x-components/src/x-installer/x-installer/x-installer.ts:310]: I needed to access to the snippetConfig in the `provide` function below. Hmm not sure if there is a better way... I thought declaring the function outside the `new vue({...})` but then I had to type it.

File: packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue [hunk 3]
```diff
@@ -73,7 +73,7 @@
      */
     @Watch('snippetConfig', { deep: true, immediate: true })
     syncExtraParams(snippetConfig: SnippetConfig): void {
-      forEach({ ...snippetConfig, ...this.values }, (name, value) => {
+      forEach({ ...this.values, ...snippetConfig }, (name, value) => {
```
[the remark this answers is not in the document]
Comment (guillei10) [reply packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue:76]: Well we decided this with @tajespasarela, and it's more about what we want to give priority to. Without this change, the component props are always overriding whatever is coming from the snippetConfig. So the new `setSnippetConfig` has no effect if the value is already passed in the props of this component.

File: packages/x-components/src/x-installer/x-installer/x-installer.ts [hunk 4]
```diff
@@ -99,6 +99,15 @@ const defaultAdapterConfig: DeepPartial<EmpathyAdapterConfig> = {
 export class XInstaller {
   private api?: XAPI;
 
+  /**
+   * Creates the public {@link XAPI} using the `api` option from {@link InstallXOptions}. If this
+   * `api` option is not passed, then a default {@link BaseXAPI} is created. To disable the API
+   * creation the value `false` must be passed in the `api` option.
+   *
+   * @internal
+   */
```
[the remark this answers is not in the document]
Comment (guillei10) [reply packages/x-components/src/x-installer/x-installer/x-installer.ts:108]: Ups

File: packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue (hunk 3 above)
[the remark this answers is not in the document]
Comment (javieri-empathy) [reply packages/x-components/src/x-modules/extra-params/components/snippet-config-extra-params.vue:76]: Sold! I hadn't thought of simply calling the new function you just added.

File: packages/x-components/src/x-installer/x-installer/x-installer.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (javieri-empathy) [reply packages/x-components/src/x-installer/x-installer/x-installer.ts:308]: I mean, why not just do this, without the `snippetConfigObservable` variable

```ts
this.snippetConfig = vue.observable(snippetConfig);
...
        provide() {
          return {
            snippetConfig: this.snippetConfig
          };
        },
...
```

**Review by javieri-empathy — DISMISSED** [review]

**Review by javieri-empathy — APPROVED** [review]

**Pull request merged**